parser/src/test/resources/schemas/TestRecordWithEnumField.avsc

```diff
@@ -0,0 +1,13 @@
+{
+  "type": "record",
+  "namespace": "com.acme",
+  "name": "TestRecordWithEnumField",
+  "doc": "nothing fancy",
+  "fields": [
+    {
+      "name": "enumField",
+      "type": "TestEnum",
+      "default": "SYMBOL_B"
```

Comment:
So the purpose of this change is to allow references to externally defined types, rather than needing to inline everything?

Reply:
allowing avsc reuse has always been the goal. this PR is mostly about completing missing functionality and fixing bugs